torch/distributed/fsdp/flat_param.py

```diff
@@ -498,6 +498,16 @@ def __init__(
         self._training_state = HandleTrainingState.IDLE
         self._debug_level = dist.get_debug_level()
         self._fully_sharded_module = fully_sharded_module
+        # NB: this is updated during lazy initialization because optimizers can be applied after
+        # wrapping with FSDP.
```

Comment:
This seems to be a duplicated one as what is in `_runtime_utils.py`. Since optimizer may be applied after wrapping with FSDP, should we consolidate to only one check?

Reply:
Consoildated to only after wrapping with FSDP check.